### PR TITLE
fix(DeleteResource): use Trans component

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
We didn't use react-i18next Trans component because of i18next-parser.
In DeleteResource container, we then had split the string into 3 part because we want to display the resource name in bold. It's harder for translation team to know how to translate the parts.

We switched to i18next-scanner that support Trans.

**What is the chosen solution to this problem?**
In DeleteResource, use Trans instead of splitting the string.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/examples/config/i18n.js [hunk 1]
```diff
@@ -8,9 +8,10 @@ i18n.init({
 	resources: {
 		en: {
 			[I18N_DOMAIN_CONTAINERS]: {
-				DELETE_RESOURCE_MESSAGE: 'Are you sure you want to remove the {{resourceLabel}}',
-				DELETE_RESOURCE_MESSAGE_female: 'Are you sure you want to remove the {{resourceLabel}}',
-				DELETE_RESOURCE_QUESTION_MARK: '?',
+				DELETE_RESOURCE_MESSAGE:
+					'Are you sure you want to remove the <1>{{resourceType}}</1> <3><0>{{label}}</0></3> ?',
```
Comment (annilsson) [inline packages/containers/examples/config/i18n.js]: Are these autogenerated? I don't understand what `<3><0>` means (two tags)?
I was looking on https://react.i18next.com/components/trans-component but didn't find any examples with two

Comment (jsomsanith) [reply packages/containers/examples/config/i18n.js]: Yeah their doc is awful.
- yes it's generated by i18next-scanner, and follows how Trans works
- see `<index>` as the index of the react element child.

`Are you sure you want to remove the <1>{{resourceType}}</1> <3><0>{{label}}</0></3> ?`
`0`: Are you sure you want to remove the
`1`: {{resourceType}}
`2`: ' ' (space)
`<3/>`: `<strong/>` element that has 1 child
`<3><0/></3>`: the first child of `<3/>` -->  {{label}}

File: packages/containers/src/DeleteResource/DeleteResource.container.js [hunk 2]
```diff
@@ -115,16 +114,19 @@ export class DeleteResource extends React.Component {
...
-			question = this.props.t('DELETE_RESOURCE_MESSAGE', {
-				defaultValue: 'Are you sure you want to remove the {{resourceLabel}}',
-				resourceLabel: resourceInfo.resourceTypeLabel,
-			});
+			question = (
+				<Trans i18nKey="DELETE_RESOURCE_MESSAGE">
+					Are you sure you want to remove the {{ resourceType: resourceInfo.resourceTypeLabel }}{' '}
+					<strong>{{ label: resourceInfo.label }}</strong> ?
+				</Trans>
+			);
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/containers/src/DeleteResource/DeleteResource.container.js]: --> prettier formatting :)

[the remark this answers is not in the document]
Comment (lmaillet) [reply packages/containers/src/DeleteResource/DeleteResource.container.js]: I agree about formatting :relaxed:

But it was mainly about using
```javascript
const resourceType = resourceInfo.resourceTypeLabel;
const label = resourceInfo.label;
```
And an lighter content in Trans.
```jsx
<Trans i18nKey="DELETE_RESOURCE_MESSAGE">
	Are you sure you want to remove the {{ resourceType }}{' '}
	<strong>{{ label }}</strong>?
</Trans>
```

[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/containers/src/DeleteResource/DeleteResource.container.js]: It's exactly the same, except that your solution creates 2 vars that are used once, using short object notation

[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/containers/src/DeleteResource/DeleteResource.container.js]: Done

File: packages/containers/examples/config/i18n.js (hunk 1 above)
Comment (annilsson) [reply packages/containers/examples/config/i18n.js]: Aah I see, thanks for explanation.

The only thing I see here is that these _types_ of translations might need to be verified by a dev before a translator touches it. For example if they change something that means the `<3>` should become a `<2>`, it's not so easy to know about that. 

But I don't think there's much to do about that on our side.

Comment (jsomsanith) [reply packages/containers/examples/config/i18n.js]: The extraction is automatic, so no worries about that.
Then, if something changes on the value, the translation team will have a `modified` status. And they need to know that the `<number>` should be the same.

**Pull request merged**